packages/kbn-observable/src/observable.ts

```diff
@@ -0,0 +1,422 @@
+   * // "unsubscribed!" after 2.5s
+   *
+   *
+   * @param {Observer|Function} observerOrNext (optional) Either an observer with methods to be called,
+   *  or the first of three possible handlers, which is the handler for each value emitted from the subscribed
+   *  Observable.
+   * @return {Subscription} a subscription reference to the registered handlers
+   * @method subscribe
+   */
+  subscribe(
+    observerOrNext: PartialObserver<T> | ((value: T) => void) = {},
+    ...rest: never[]
+  ) {
+    if (rest.length > 0) {
+      throw new TypeError(
+        'Specifying multiple arguments to "subscribe" is not allowed. If you ' +
+          'want to specify "error" or "complete" handler, pass in an object ' +
+          'instead.'
+      );
+    }
+
+    const observer: PartialObserver<T> =
+      typeof observerOrNext === 'function'
+        ? { next: observerOrNext }
+        : observerOrNext;
+
+    return new Subscription(observer, this._onSubscribe);
+  }
+
+  pipe(): Observable<T>;
+  pipe<A>(op1: OperatorFunction<T, A>): Observable<A>;
+  pipe<A, B>(
+    op1: OperatorFunction<T, A>,
+    op2: OperatorFunction<A, B>
+  ): Observable<B>;
+  pipe<A, B, C>(
+    op1: OperatorFunction<T, A>,
+    op2: OperatorFunction<A, B>,
+    op3: OperatorFunction<B, C>
+  ): Observable<C>;
+  pipe<A, B, C, D>(
+    op1: OperatorFunction<T, A>,
+    op2: OperatorFunction<A, B>,
+    op3: OperatorFunction<B, C>,
+    op4: OperatorFunction<C, D>
+  ): Observable<D>;
+  pipe<A, B, C, D, E>(
+    op1: OperatorFunction<T, A>,
+    op2: OperatorFunction<A, B>,
+    op3: OperatorFunction<B, C>,
+    op4: OperatorFunction<C, D>,
+    op5: OperatorFunction<D, E>
+  ): Observable<E>;
+  pipe<A, B, C, D, E, F>(
+    op1: OperatorFunction<T, A>,
+    op2: OperatorFunction<A, B>,
+    op3: OperatorFunction<B, C>,
+    op4: OperatorFunction<C, D>,
+    op5: OperatorFunction<D, E>,
+    op6: OperatorFunction<E, F>
+  ): Observable<F>;
+  pipe<A, B, C, D, E, F, G>(
+    op1: OperatorFunction<T, A>,
+    op2: OperatorFunction<A, B>,
+    op3: OperatorFunction<B, C>,
+    op4: OperatorFunction<C, D>,
+    op5: OperatorFunction<D, E>,
+    op6: OperatorFunction<E, F>,
+    op7: OperatorFunction<F, G>
+  ): Observable<G>;
+  pipe<A, B, C, D, E, F, G, H>(
+    op1: OperatorFunction<T, A>,
+    op2: OperatorFunction<A, B>,
+    op3: OperatorFunction<B, C>,
+    op4: OperatorFunction<C, D>,
+    op5: OperatorFunction<D, E>,
+    op6: OperatorFunction<E, F>,
+    op7: OperatorFunction<F, G>,
+    op8: OperatorFunction<G, H>
+  ): Observable<H>;
+  pipe<A, B, C, D, E, F, G, H, I>(
+    op1: OperatorFunction<T, A>,
+    op2: OperatorFunction<A, B>,
+    op3: OperatorFunction<B, C>,
+    op4: OperatorFunction<C, D>,
+    op5: OperatorFunction<D, E>,
+    op6: OperatorFunction<E, F>,
+    op7: OperatorFunction<F, G>,
+    op8: OperatorFunction<G, H>,
+    op9: OperatorFunction<H, I>
+  ): Observable<I>;
+  pipe<R>(...operations: OperatorFunction<T, R>[]): Observable<R>;
```

Comment:
Yep, the method itself is not included. It's described in the last paragraph here: https://www.typescriptlang.org/docs/handbook/functions.html#overloads

> Note that the function `pickCard(x): any` piece is not part of the overload list, so it only has two overloads: one that takes an object and one that takes a number. Calling `pickCard` with any other parameter types would cause an error.

Reply:
Ah, good to know, thanks!